src/main/java/hudson/plugins/performance/actions/PerformanceProjectAction.java

```diff
@@ -388,29 +390,35 @@ public void doRespondingTimeGraphPerTestCaseMode(
                 if (!buildsLimits.includedByStep(build.number)) {
                     continue;
                 }
-                PerformanceBuildAction performanceBuildAction = build
-                        .getAction(PerformanceBuildAction.class);
-                if (performanceBuildAction == null) {
-                    continue;
-                }
-                PerformanceReport performanceReport = performanceBuildAction
-                        .getPerformanceReportMap().getPerformanceReport(
-                                performanceReportNameFile);
+                PerformanceReport performanceReport = getPerformanceReport(build, performanceReportNameFile);
                 if (performanceReport == null) {
                     nbBuildsToAnalyze--;
                     continue;
                 }
 
                 List<UriReport> uriListOrdered = performanceReport.getUriListOrdered();
                 for (UriReport uriReport : uriListOrdered) {
-                    dataSetBuilderAverage.add(uriReport.getAverage(), uriReport.getUri(), label);
+                    dataSetBuilder.add(valueSelector.getValue(uriReport), uriReport.getUri(), label);
                 }
             }
             nbBuildsToAnalyze--;
         }
         ChartUtil.generateGraph(request, response,
-                createRespondingTimeChart(dataSetBuilderAverage.build()), 600, 200);
+                createRespondingTimeChart(dataSetBuilder.build()), 600, 200);
+    }
 
+    protected PerformanceReport getPerformanceReport(Run<?, ?> build, String reportFileName) {
+        PerformanceBuildAction performanceBuildAction = build.getAction(PerformanceBuildAction.class);
+        if (performanceBuildAction == null) {
+            return null;
```

Comment:
slight change in behaviour: `nbBuildsToAnalyze--;` now runs when the build has no PerformanceBuildAction. is this correct?